pkg/network/ebpf/c/protocols/http/buffer.h

```diff
@@ -33,6 +33,22 @@ static __always_inline void read_into_buffer(char *buffer, char *data, size_t da
     }
 }
 
+static __always_inline void read_into_buffer_classification(char *buffer, char *data, size_t data_size) {
+    bpf_memset(buffer, 0, CLASSIFICATION_MAX_BUFFER);
+
+    // we read CLASSIFICATION_MAX_BUFFER-1 bytes to ensure that the string is always null terminated
+    if (bpf_probe_read_user_with_telemetry(buffer, CLASSIFICATION_MAX_BUFFER - 1, data) < 0) {
```

Comment:
are we always expecting a string for classification buffer? if yes, consider using bpf_probe_read_user_str_with_telemetry, I think it is a bit more [efficient](https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h#L3998)

Reply:
no, we have couple of binary protocols
and `bpf_probe_read_user_str` expects to have `null` in the end of the string, which is not the use case (even for http classification)